systems/WikibaseRepo/04-Solution_Strategy.md

```diff
@@ -1,16 +1,16 @@
 # Solution Strategy
 
-## Developing Wikibase Repo as a MediaWiki extension
+## Developing Wikibase Repository as a MediaWiki extension
 
-One of the main requirements of Wikibase is to make data available to [MediaWiki](../overview/12-Glossary.md#mediawiki) applications. This makes it a natural choice to deliver the client (= data consumer) functionality as a MediaWiki extension. The same is not necessarily the case for the Wikibase Repository, the subsystem that provides and maintains the structured data. WikibaseRepo could have been developed independently of MediaWiki, yet it was decided to use MediaWiki like a web framework and add the functionality to store and maintain structured data through the WikibaseRepo MediaWiki extension.
+One of the main requirements of Wikibase is to make data available to [MediaWiki](../overview/12-Glossary.md#mediawiki) applications. This makes it a natural choice to deliver the Wikibase Client (= data consumer) functionality as MediaWiki extensions. The same is not necessarily the case for the Wikibase Repository, the subsystem that provides and maintains the structured data. Wikibase Repository could have been developed independently of MediaWiki, yet it was decided to use MediaWiki like a web framework and add the functionality to store and maintain structured data through the MediaWiki extensions.
```

Comment:
This paragraph now manages to describe the reasons for "Developing Wikibase Repository as a MediaWiki extension" without even mentioning the WikibaseRepo extension. The last sentence is really odd, as it says the structured data functionality was added "through the MediaWiki extensions" (which ones?) and is also in contrast to the headline, which says "Wikibase Repository **as a Mediawiki extension**".

Maybe: "Wikibase Repository could have been developed independently of MediaWiki, yet it was decided to use MediaWiki like a web framework and add the functionality to store and maintain structured data through the WikibaseRepo MediaWiki extension, optionally combined with additional Wikibase extensions such as WikibaseLexeme." (with correct Glossary links)

Reply:
> "through the MediaWiki extensions" (which ones?)

So "Wikibase Client" ends up being made up of multiple extensions including ArticlePlaceholder, InterwikiSorting? etc, none of which are really reflected in the docs yet as we didn't get there.

Already one improvement there might be to say "as a set of MediaWiki extensions".
But that still sounds a bit odd.

This should probably try to reflect that the core functionality is provided in a single MediaWiki extension, being WikibaseClient, but within the whole Wikibase Client there are other things included too.

Reply:
Headline needs fixing and the "the" in the final sentence needs removing